src/messages/serialize.rs

```diff
@@ -0,0 +1,64 @@
+//! Serialization rules specified in [RFC-001#serializationdeserialization]
+//!
+//! We automatically serialize and deserialize using serde derivations where possible.
+//! Sometimes we need to implement ourselves, this file holds that code.
+//!
+//! [RFC-001#rules]: https://github.com/ZcashFoundation/redjubjub/blob/main/rfcs/0001-messages.md#serializationdeserialization
+
+use serde::ser::{Serialize, Serializer};
+
+use serde::de::{self, Deserialize, Deserializer, Visitor};
+
+use super::constants::{AGGREGATOR_PARTICIPANT_ID, DEALER_PARTICIPANT_ID};
+use super::*;
+
+use std::fmt;
+
+impl Serialize for ParticipantId {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        match *self {
+            ParticipantId::Signer(id) => serializer.serialize_u8(id),
```

Comment:
Missing check:
> Must be less than or equal to `MAX_SIGNER_PARTICIPANT_ID`.

Reply:
Since you're serializing data you control, you can implement this check using `assert!()`.